workers.py

```diff
@@ -72,46 +70,31 @@ def worker() -> None:
             )
             prometheus_metrics.METRICS['get_errors'].inc()
             return
-
         prometheus_metrics.METRICS['get_successes'].inc()
 
-        try:
-            with NamedTemporaryFile(delete=False) as tmp_file:
-                file_name = tmp_file.name
-
-                for chunk in filter(None, resp.iter_content(chunk_size=CHUNK)):
-                    tmp_file.write(chunk)
+        file_obj = BytesIO(resp.content)
 
-        except IOError as exception:
-            logger.error(
-                '%s: Unable to create temp file for "%s": %s',
-                thread.name, source_id, exception
-            )
-            return
-
-        # Unpack data and stream it
-
-        # Build the POST data object
-        data = {
-            'id': source_id,
-            'data': custom_parser.parse(file_name),
+        # Store payload ID in a header
+        headers = {
+            'source_id': source_id,
         }
 
         # Pass to next service
         prometheus_metrics.METRICS['posts'].inc()
         try:
-            resp = _retryable('post', f'http://{dest_url}', json=data)
-            prometheus_metrics.METRICS['post_successes'].inc()
+            resp = _retryable(
+                'post',
+                f'http://{dest_url}',
+                headers=headers,
+                data=custom_parser.parse(file_obj)
+            )
         except requests.HTTPError as exception:
             logger.error(
                 '%s: Failed to pass data for "%s": %s',
                 thread.name, source_id, exception
             )
             prometheus_metrics.METRICS['post_errors'].inc()
-
-        # Cleanup
-        with suppress(IOError):
-            os.remove(file_name)
+        prometheus_metrics.METRICS['post_successes'].inc()
```

Comment:
This line needs to be moved just below -
```python
resp = _retryable(
                'post',
                f'http://{dest_url}',
                headers=headers,
                data=custom_parser.parse(file_obj)
            )
```
inside the `try`, that is.
otherwise in case of a POST failure, the counter for POST success would also be incremented